sql/api/src/main/scala/org/apache/spark/sql/catalyst/parser/parsers.scala

```diff
@@ -249,20 +249,24 @@ class ParseException private(
   override def getMessage: String = {
     val builder = new StringBuilder
     builder ++= "\n" ++= message
-    start match {
-      case Origin(Some(l), Some(p), _, _, _, _, _, _, _) =>
-        builder ++= s" (line $l, pos $p)\n"
-        command.foreach { cmd =>
-          val (above, below) = cmd.split("\n").splitAt(l)
-          builder ++= "\n== SQL ==\n"
-          above.foreach(builder ++= _ += '\n')
-          builder ++= (0 until p).map(_ => "-").mkString("") ++= "^^^\n"
-          below.foreach(builder ++= _ += '\n')
-        }
-      case _ =>
-        command.foreach { cmd =>
-          builder ++= "\n== SQL ==\n" ++= cmd
-        }
+    if (queryContext.nonEmpty) {
+      builder ++= queryContext.head.summary()
```

Comment:
Shall we append all the summaries from `queryContext` instead of just the first query context?

Reply:
Oh, prep for the future? Or would this actually work? If so we should generalize this.

Reply:
Yes, prep for the future. We can simply write `builder ++= queryContext.map(_.summary()).mkString("\n")` here